examples/rules_ios/Lib/BUILD

```diff
@@ -0,0 +1,44 @@
+load(
+    "@build_bazel_rules_ios//rules:framework.bzl",
+    rules_ios_apple_framework = "apple_framework",
+)
+
+exports_files(
+    ["Info.plist"],
+    visibility = ["//Lib:__subpackages__"],
+)
+
+rules_ios_apple_framework(
+    name = "Lib",
+    srcs = ["Resources.swift"],
+    bundle_id = "rules-xcodeproj.LibFramework",
+    infoplists = ["Info.plist"],
+    module_name = "Lib",
+    platforms = {
+        "ios": "15.0",
+    },
+    visibility = [
+        "//Lib/dist/dynamic:__pkg__",
+        "//UI:__pkg__",
+        "//WidgetExtension:__subpackages__",
+    ],
+    xcconfig = {
+        "APPLICATION_EXTENSION_API_ONLY": "YES",
+    },
+)
+
+rules_ios_apple_framework(
+    name = "LibDynamic",
+    bundle_id = "rules-xcodeproj.LibFramework",
+    infoplists = ["//Lib:Info.plist"],
+    link_dynamic = True,
+    module_name = "Lib",
+    platforms = {
+        "ios": "15.0",
+    },
+    visibility = ["@rules_xcodeproj//xcodeproj:generated"],
+    xcconfig = {
```

Comment:
Do we want to test the `_by_build_setting` ones for this and infoplists

Reply:
You want to add it in a follow up, @luispadron? We don't use that internally so I'm less familiar with its usage.

Reply:
Yep can do!

Reply:
The `_by_build_setting` attrs are loosely tied to intended usage of `--config` which is a bit different then how `rules_xcodeproj` creates "Xcode Configurations". I think it makes sense to tackle that separately, ideally we would not only cover the `_by_build_setting` attributes but also how to have `rules_ios` run with `rules_xcodeproj`'s "Xcode Configurations" feature.

Probably a good segway to revisit the `_by_build_setting` story in `rules_ios` and see if there's any improvements we can do there to make it more generic and inclusive of other Xcode project generators, it was initially closely related to `rules_ios`'s own generator.